### PR TITLE
feat: add abilty to edit memories

### PR DESCRIPTION
**Pull request by Peter-De-Ath**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [x] **Testing:** Have you written and run sufficient tests for validating the changes?
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description
Allow updating of memories

### Added
Add the ability to update a memory via the API using a PATCH request.
Add ability to update a memory in the UI.

### Changed
Update memories list heading to "Last Modified" from "Created At"
Updated timestamp to include the hours mins and seconds

### Additional Information

- [Insert any additional context, notes, or explanations for the changes]
  - [Reference any related issues, commits, or other relevant information]
https://github.com/open-webui/open-webui/issues/2729

### Screenshots or Videos

- [Attach any relevant screenshots or videos demonstrating the changes]
image
image

### CODE REVIEW

File: backend/apps/webui/models/memories.py [hunk 1]
```diff
@@ -64,6 +64,20 @@ def insert_new_memory(
             return memory
         else:
             return None
+        
+    def update_memory(
```
Comment (tjbck) [inline backend/apps/webui/models/memories.py:68]: `update_memory_by_id` would be more descriptive here

File: backend/apps/webui/routers/memories.py [hunk 2]
```diff
@@ -62,6 +64,27 @@ async def add_memory(
     return memory
 
 
+@router.patch("/{memory_id}", response_model=Optional[MemoryModel])
```
Comment (tjbck) [inline backend/apps/webui/routers/memories.py:67]: `post` with `/update` endpoint should be used here for convention

File: src/lib/components/chat/Settings/Personalization/AddMemoryModal.svelte [hunk 3]
```diff
@@ -2,21 +2,60 @@
...
-	import { addNewMemory } from '$lib/apis/memories';
+	import { addNewMemory, updateMemoryById } from '$lib/apis/memories';
 	import { toast } from 'svelte-sonner';
 
 	const dispatch = createEventDispatcher();
 
 	export let show;
+	export let memory = {};
+
+	let showUpdateBtn = false;
```
Comment (tjbck) [inline src/lib/components/chat/Settings/Personalization/AddMemoryModal.svelte:13]: I'll update the frontend part!

**Review by tjbck — CHANGES_REQUESTED** [review]

File: src/lib/components/chat/Settings/Personalization/AddMemoryModal.svelte (hunk 3 above)
Comment (Peter-De-Ath) [reply src/lib/components/chat/Settings/Personalization/AddMemoryModal.svelte:13]: Yes Sir! :+1: 
Appreciate that, first time working with svelte

**Comment by tjbck** [comment]
I'll merge this first then refactor, LGTM, Thanks!

**Pull request merged**